### PR TITLE
Adding domT method for component interpolation

### PR DESCRIPTION
**Pull request by berhalak**
New function domT that is addressing this issue https://github.com/gristlabs/grist-core/issues/313.
It allows using components as values for translations that require string interpolation.

### CODE REVIEW

File: app/client/lib/localization.ts [hunk 1]
```diff
@@ -68,14 +70,48 @@ export async function setupLocale() {
...
+/**
+ * Resolves the translation of the given key and substitutes. Supports dom elements interpolation.
+ */
+ export function domT(key: string, args?: any, instance = i18next): DomContents {
+  if (!instance.exists(key)) {
+    const error = new Error(`Missing translation for key: ${key} and language: ${i18next.language}`);
+    reportError(error);
+  }
+  // If there are any DomElements in args, handle it with missingInterpolationHandler.
+  const domElements = Object.entries(args).filter(([_, value]) => isLikeDomContents(value));
```
Comment (paulfitz) [inline app/client/lib/localization.ts:93]: If `isLikeDomContents` is inexpensive, would it be reasonable to just rename your `domT` to `t`, and current `t` to `translateWithoutInterpolation`? So translator can generally just always use the same `t` function unless there is some very specific reason not to consider interpolating?

**Comment by paulfitz** [comment]
@arnaudpeich does this PR look like it does everything you need for #313?

File: app/client/lib/localization.ts (hunk 1 above)
Comment (berhalak) [reply app/client/lib/localization.ts:93]: Yes I'm happy to switch it, I started with that idea initially, but then I had some doubts, as it won't be consistent with the server side. But I think this is what we want most of the time on client side.

**Comment by arnaudpeich** [comment]
> @arnaudpeich does this PR look like it does everything you need for #313?

Thanks for the ping! I haven't looked at the implementation, but based on the example, yes this is exactly what we're looking for 🙌🏻

File: app/client/lib/localization.ts (hunk 1 above)
Comment (paulfitz) [reply app/client/lib/localization.ts:93]: I see. Yes, I think this would be a reasonable kind of inconsistency.

Comment (berhalak) [reply app/client/lib/localization.ts]: I've included simple type inference, so for plain values the result should be a string. @arnaudpeich I also included a helper method `makeT` that should be helpful and can eliminate the need to create `translate` helpers by hand.

Comment (arnaudpeich) [reply app/client/lib/localization.ts]: Looks great, thank you! I'll update our PR accordingly :)

**Review by paulfitz — APPROVED** [review]
Looks like it will do the job. Thanks @berhalak !

**Pull request merged**